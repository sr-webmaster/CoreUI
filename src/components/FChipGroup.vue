<template>
  <v-card class="f-chip-group">
    <v-card-title v-if="hasTitle">
      {{ title }}
    </v-card-title>
    <v-divider v-if="hasTitle" />
    <v-card-text>
      <f-chip
        v-for="item in tagsComputed"
        :key="item.id"
        :color="isSelected(item.id) ? 'primary' : 'accent'"
        @click.prevent="changeTags(item)"
      >
        {{ item.name }}
      </f-chip>
    </v-card-text>
  </v-card>
</template>

<script>
import remove from 'lodash/remove'
import FChip from './FChip'

export default {
  components: {
    FChip
  },
  props: {
    title: {
      type: String,
      default: ''
    },
    tags: {
      type: Array,
      required: true
    },
    selected: {
      type: Array,
      default: () => []
    }
  },
  data () {
    return {
      selectedTagsData: [...this.selected]
    }
  },
  computed: {
    hasTitle () {
      return this.title.length > 0
    },
    tagsComputed () {
      return this.tags
    }
  },
  watch: {
    selected (val) {
      this.selectedTagsData = val
    }
  },
  methods: {
    changeTags (tag) {
      if (!this.isSelected(tag.id)) {
        this.selectedTagsData.push(tag)
      } else {
        this.selectedTagsData = remove(this.selectedTagsData, function (item) {
          return !(item.id === tag.id)
        })
      }
      this.$emit('input', this.selectedTagsData)
    },
    isSelected (id) {
      return this.selectedTagsData.some(element => element.id === id)
    }
  }
}
</script>

<style lang="scss">
@import '@freshinup/core-ui/src/styles/sass/_variables';
.title {
  font-size: $fresh-typography--title-link !important;
}
.title:link {
  text-decoration: none;
}
</style>
