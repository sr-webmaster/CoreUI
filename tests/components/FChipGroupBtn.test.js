import { mount, shallowMount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FChipGroupBtn.vue'
import { FIXTURE_UI_FCHIPGROUP_TAGS } from '@freshinup/core-ui/tests/__data__/FChipGroup/tags'

describe('FChipGroupBtn', () => {
  describe('Visuals', () => {
    test('setting required properties', async () => {
      const wrapper = mount(Component, {
        propsData: {
          title: 'Button Title',
          selected: [],
          tags: FIXTURE_UI_FCHIPGROUP_TAGS
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    test('input() emits input with value', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          title: 'Button Title',
          tags: FIXTURE_UI_FCHIPGROUP_TAGS
        }
      })
      wrapper.vm.input('tag 1')
      // assert event has been emitted
      expect(wrapper.emitted('input')).toBeTruthy()
      expect(wrapper.emitted('input')).toHaveLength(1)
      expect(wrapper.emitted('input')[0]).toEqual(['tag 1'])
    })
  })
})
