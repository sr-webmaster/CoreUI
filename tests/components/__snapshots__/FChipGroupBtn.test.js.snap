// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FChipGroupBtn Visuals setting required properties 1`] = `
<div
  class="v-menu f-chip-group-btn"
  width="300"
>
  <button
    class="f-chip-group-btn__activator v-btn theme--light undefined-text--text"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
      Button Title
      
      <!---->
    </div>
  </button>
  <div
    class="v-menu__content theme--light "
    style="max-height: auto; min-width: 0px; max-width: 300px; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
  >
     
    <div
      class="f-chip-group v-card v-sheet theme--light"
    >
      <!---->
       
      <!---->
       
      <div
        class="v-card__text"
      >
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 1
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 2
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 3
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 4
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 5
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 6
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 7
    
          </span>
        </span>
        <span
          class="v-chip theme--light accent accent-text--text"
          tabindex="0"
        >
          <span
            class="v-chip__content"
          >
            
      Category 8
    
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
